consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java

```diff
@@ -18,382 +18,393 @@
-			for(Map.Entry<String, JsonNode> sourceDocument : sourceDocuments.entrySet()) {
-				List<Query> docConditions = new LinkedList<>();
-				for(String keyField : getJobConfiguration().getDestinationEntityKeyFields()) {
-					ValueQuery docQuery = new ValueQuery(keyField + " = " + sourceDocument.getValue().findValue(keyField).asText());
-					docConditions.add(docQuery);	
-				}	
-				requestConditions.add(and(docConditions));
-			}
-			destinationRequest.where(or(requestConditions));
-			destinationRequest.select(includeFieldRecursively("*"));
-			destinationRequest.sort(new SortCondition(getJobConfiguration().getSourceTimestampPath(), SortDirection.ASC));
-			destinationDocuments = findDestinationData(destinationRequest);
-		} catch (IOException e) {
-			LOGGER.error("Error getting destinationDocuments", e);
-		}
-		return destinationDocuments;
-	}
-
-	protected List<JsonNode> getDocumentsToOverwrite(Map<String, JsonNode> sourceDocuments, Map<String, JsonNode> destinationDocuments) {
-		List<JsonNode> documentsToOverwrite = new ArrayList<>();
-		for(Map.Entry<String, JsonNode> sourceDocument : sourceDocuments.entrySet()) {
-			JsonNode destinationDocument = destinationDocuments.get(sourceDocument.getKey());
-			if(destinationDocument == null) {
-				documentsToOverwrite.add(sourceDocument.getValue());
-			} else if (!documentsConsistent(sourceDocument.getValue(), destinationDocument)) {
-				documentsToOverwrite.add(sourceDocument.getValue());
-			}
-		}
-		return documentsToOverwrite;
-	}
-	
-	protected boolean documentsConsistent(JsonNode sourceDocument, JsonNode destinationDocument) {
-		boolean consistent = true;
-		
-		Iterator<Entry<String, JsonNode>> nodeIterator = sourceDocument.fields();
-		
-		while (nodeIterator.hasNext()) {
-			Entry<String, JsonNode> sourceNode = nodeIterator.next();
-			
-			if(!getJobConfiguration().getComparisonExclusionPaths().contains(sourceNode.getKey())) {
-				if(!sourceNode.getValue().equals(destinationDocument.findValue(sourceNode.getKey()))) {
-					consistent = false;
-					break;
-				}	
-			}
-		}
-		return consistent;
-	}
-	
-	protected Map<String, JsonNode> findSourceData(LightblueRequest findRequest) throws IOException {
-		return getJsonNodeMap(getSourceClient().data(findRequest, JsonNode[].class), getJobConfiguration().getDestinationEntityKeyFields());
-	}
-	
-	protected Map<String, JsonNode> findDestinationData(LightblueRequest findRequest) throws IOException {
-		return getJsonNodeMap(getDestinationClient().data(findRequest, JsonNode[].class), getJobConfiguration().getDestinationEntityKeyFields());
-	}
-
-	protected LinkedHashMap<String, JsonNode> getJsonNodeMap(JsonNode[] results, List<String> entityKeyFields) {
-		LinkedHashMap<String, JsonNode> resultsMap = new LinkedHashMap<>();
-		for(JsonNode result : results) {
-			StringBuilder resultKey = new StringBuilder();
-			for(String keyField : entityKeyFields) {
-				resultKey.append(result.findValue(keyField)).append("|||");
```

Comment:
I can no longer reproduce so I  reverted that particular commit. If it comes up again, we can tackle it in another issue.